File: backend/src/main/java/com/travelcompanion/controller/WeatherController.java
package com.travelcompanion.controller;

import com.travelcompanion.service.WeatherService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/weather")
@RequiredArgsConstructor
public class WeatherController {

    private final WeatherService weatherService;

    @GetMapping("/{city}")
    public ResponseEntity<Map<String, Object>> getWeather(@PathVariable String city) {
        return ResponseEntity.ok(weatherService.getWeather(city));
    }

    @GetMapping("/{city}/forecast")
    public ResponseEntity<Map<String, Object>> getForecast(@PathVariable String city) {
        return ResponseEntity.ok(weatherService.getForecast(city));
    }
}

File: backend/src/controllers/weather.controller.js
const weatherService = require('../services/weather.service');

exports.getWeather = async (req, res, next) => {
  try {
    const weather = await weatherService.getWeather(req.params.city);
    res.json(weather);
  } catch (error) {
    next(error);
  }
};

exports.getForecast = async (req, res, next) => {
  try {
    const forecast = await weatherService.getForecast(req.params.city);
    res.json(forecast);
  } catch (error) {
    next(error);
  }
};
